src/deploy/osp_deployer/director.py

```diff
@@ -846,6 +855,57 @@ def setup_nic_configuration(self):
         for cmd in cmds:
             self.run(cmd)
 
+    def setup_dpdk_nic_configuration(self):
+
+        logger.debug("setting ovs dpdk environment")
+        # Get the user supplied NIC settings from the .ini
+        ini_nics_settings = self.settings.get_curated_nics_settings()
+
+        cmds = []
+        dpdk_conf = {}
+        env_file = os.path.join(self.templates_dir, "neutron-ovs-dpdk.yaml")
+
+        # Get and sort the Dpdk interfaces that the user configured
+        for setting_name, setting_value in ini_nics_settings.iteritems():
+            if setting_name.find('Dpdk') != -1:
+                dpdk_conf.update({setting_name: setting_value})
+        dpdk_interfaces = [x[1] for x in sorted(dpdk_conf.items())]
+
+        # The following is joining only the first two dpdk interfaces
+        # for mode 2 or all the interfaces (4) for mode 1
+        if self.settings.ovs_dpdk_mode == 2:
+            interfaces = "'" + ",".join(dpdk_interfaces[0:2]) + "'"
+        else:
+            interfaces = "'" + ",".join(dpdk_interfaces) + "'"
+
+        # Build up the sed command to perform variable substitution
+        # in the neutron-ovs-dpdk.yaml (dpdk environment)
+        cmds.append('sed -i "s|DpdkInterfaces:.*|DpdkInterfaces: ' +
+                    interfaces + '|" ' + env_file)
+
+        nic_config_dirname = os.path.dirname(self.settings.nic_env_file)
```

Comment:
I don't think lines 886-903 are needed since the nic_env_file will already be pointing at the correct compute.yaml file.

Reply:
I don't think so. The name of the compute.yaml will changed based on the mode enabled. For example if mode 1 is chosen then it needs to point to compute-ovs-dpdk-mode1.yaml, if mode 2 is chosen then compute-ovs-dpdk-mode2.yaml will be picked up.